### PR TITLE
TRITON-853 cloudapi should allow creation of instances with delegated datasets

### PR DESCRIPTION
**Pull request by timfoster**

### CODE REVIEW

**Comment by bahamat** [comment]
See also joyent/sdc-cloudapi#64

**Comment by bahamat** [comment]
I've also updated this now to reflect @twhiteman's current changes to cloudapi.

File: lib/do_instance/do_create.js [hunk 1]
```diff
@@ -496,6 +498,19 @@ do_create.options = [
...
+        type: 'bool',
+        help: 'Create a Delegated Dataset on this instance. If set a ' +
+              'delegated dataset is created, though the instance is ' +
+              'prevented from receiving ZFS datasets (zfs recv). Note ' +
+              'that Triton CloudAPI instances must have the SAPI config ' +
+              'value "experimental_cloudapi_delegate_dataset=true" ' +
+              'for create requests using this option to be allowed.',
+        // mark this option hidden for now, see TRITON-853, at least until we
+        // get a full implementation of RFD 044.
+        hidden: true
```
Comment (twhiteman) [inline lib/do_instance/do_create.js:512]: Just checking - is this still the case (that this option should be hidden)?

Comment (bahamat) [reply lib/do_instance/do_create.js:512]: I have no strong opinion about it. I'm ok with it not being hidden, but others may feel differently.

Comment (twhiteman) [reply lib/do_instance/do_create.js:512]: Remove hidden - CloudAPI will have it exposed, so node-triton might as well too.

Comment (bahamat) [reply lib/do_instance/do_create.js]: Done and done.

**Pull request merged**